libs/designsystem/src/lib/components/segmented-control/segmented-control.component.spec.ts

```diff
@@ -27,92 +27,129 @@ describe('SegmentedControlComponent', () => {
       expect(component.isDefaultMode).toBeTruthy();
       expect(component.isChipMode).toBeFalsy();
     });
 
-    it('should have 2 segments buttons', () => {
-      expect(fixture.nativeElement.querySelectorAll('ion-segment-button').length).toBe(2);
+    it('should have a segment button per item', () => {
+      expect(spectator.queryHostAll('ion-segment-button').length).toBe(items.length);
     });
 
     it('should not have any segmented chips', () => {
-      expect(fixture.nativeElement.querySelectorAll('kirby-chip').length).toBe(0);
+      expect(spectator.queryHostAll('kirby-chip').length).toBe(0);
     });
 
-    it('should call onSegmentSelect when ionChange event fires', () => {
+    it('should call onSegmentSelect when ionChange event fires', async () => {
+      expect(component.activeSegment).toBe(items[1]);
+      const ionSegmentElement = spectator.queryHost<HTMLIonSegmentElement>('ion-segment');
+      await TestHelper.whenHydrated(ionSegmentElement);
       spyOn(component, 'onSegmentSelect');
-      const segmentElm = fixture.debugElement.query(By.css('ion-segment'));
-      segmentElm.triggerEventHandler(
-        'ionChange',
-        new CustomEvent('ionChange', { detail: { value: items[1] } })
-      );
-      expect(component.onSegmentSelect).toHaveBeenCalledWith(items[1]);
+      const changeEvent = new CustomEvent('ionChange', { detail: { value: items[0] } });
+      ionSegmentElement.dispatchEvent(changeEvent);
+      expect(component.onSegmentSelect).toHaveBeenCalledWith(items[0]);
     });
 
-    it('should set activeSegment to second segmentItem when ionChange event fires', () => {
-      const segmentElm = fixture.debugElement.query(By.css('ion-segment'));
-      segmentElm.triggerEventHandler(
-        'ionChange',
-        new CustomEvent('ionChange', { detail: { value: items[1] } })
-      );
+    it('should set activeSegment to event.detail.value when ionChange event fires', async () => {
       expect(component.activeSegment).toBe(items[1]);
+      const ionSegmentElement = spectator.queryHost<HTMLIonSegmentElement>('ion-segment');
+      await TestHelper.whenHydrated(ionSegmentElement);
+      const changeEvent = new CustomEvent('ionChange', { detail: { value: items[2] } });
+      ionSegmentElement.dispatchEvent(changeEvent);
+      expect(component.activeSegment).toBe(items[2]);
+    });
+
+    describe('when updating items', () => {
+      it('should not emit segmentSelect event', async () => {
+        const ionSegmentElement = spectator.queryHost<HTMLIonSegmentElement>('ion-segment');
+        await TestHelper.whenHydrated(ionSegmentElement);
+
+        const clonedItems = JSON.parse(JSON.stringify(items));
+        spectator.setHostInput({ items: clonedItems });
+        expect(onSegmentSelectSpy).not.toHaveBeenCalled();
+      });
     });
   });
 
   describe('chip mode', () => {
     beforeEach(() => {
       component.mode = 'chip';
-      fixture.detectChanges();
+      spectator.detectChanges();
```

Comment:
Consider calling `spectator.setInput('mode', 'chips');` instead (performs `detectChanges` under the hood).

Reply:
Fixed